src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -4,1096 +4,686 @@
-            // the profile stub by giving it a guid so we can call _FindMatchingProfile
-            for (auto& profileStub : fullFile[JsonKey(ProfilesKey)])
+            for (const auto& schemeJson : schemes)
             {
-                if (profileStub.isMember(JsonKey(UpdatesKey)))
+                if (schemeJson.isObject())
                 {
-                    // This stub is meant to be a modification to an existing profile,
-                    // try to find the matching profile
-                    profileStub[JsonKey(GuidKey)] = profileStub[JsonKey(UpdatesKey)];
-                    auto matchingProfile = _FindMatchingProfile(profileStub);
-                    if (matchingProfile)
+                    if (const auto scheme = ColorScheme::FromJson(schemeJson))
                     {
-                        try
-                        {
-                            // We found a matching profile, create a child of it and put the modifications there
-                            // (we add a new inheritance layer)
-                            auto childImpl{ matchingProfile->CreateChild() };
-                            childImpl->LayerJson(profileStub);
-                            childImpl->Origin(OriginTag::Fragment);
-
-                            // replace parent in _profiles with child
-                            _allProfiles.SetAt(_FindMatchingProfileIndex(matchingProfile->ToJson()).value(), *childImpl);
-                        }
-                        catch (...)
-                        {
-                        }
-                    }
-                }
-                else
-                {
-                    // This is a new profile, check that it meets our minimum requirements first
-                    // (it must have at least a name)
-                    if (profileStub.isMember(JsonKey(NameKey)))
-                    {
-                        try
-                        {
-                            auto newProfile = Profile::FromJson(profileStub);
-                            // Make sure to give the new profile a source, then we add it to our list of profiles
-                            // We don't make modifications to the user's settings file yet, that will happen when
-                            // _AppendDynamicProfilesToUserSettings() is called later
-                            newProfile->Source(source);
-                            newProfile->Origin(OriginTag::Fragment);
-                            _allProfiles.Append(*newProfile);
-                        }
-                        catch (...)
-                        {
-                        }
+                        settings.globals->AddColorScheme(*scheme);
                     }
                 }
             }
         }
+    }
 
-        if (fullFile.isMember(JsonKey(SchemesKey)))
+    // profiles.defaults
+    {
+        settings.baseLayerProfile = Profile::FromJson(defaultsObject);

```

Comment:
This works even if `defaults` has been deleted (and is therefore a json `null`), correct?

Reply:
Yeah most tests don't specify a defaults object.